app/controllers/api/vcs.php

```diff
@@ -238,11 +243,22 @@
             $queueForBuilds
                 ->setType(BUILD_TYPE_DEPLOYMENT)
                 ->setResource($function)
-                ->setDeployment($deployment);
+                ->setDeployment($deployment)
+                ->setProject($project); // set the project because it won't be set for git deployments
+
+            $queueForBuilds->trigger(); // must trigger here so that we create a build for each function
 
             //TODO: Add event?
+        } catch (Throwable $e) {
+            $errors[] = $e->getMessage();
         }
     }
+
+    $queueForBuilds->setType(''); // prevent shutdown hook from triggering again
+
+    if (!empty($errors)) {
+        throw new Exception(Exception::GENERAL_UNKNOWN, \implode("\n", $errors));
```

Comment:
I think we can throw server error here, as its not public facing anyway.

Reply:
Doing that also ensures we get it on Sentry

Reply:
This should end up in Sentry since `GENERAL_UNKNOWN` is a 500 error right?